pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -27,52 +27,50 @@ import (
 )
 
 var (
-	log                     = logger.Log
-	ResultNormal            = common.ResultNormal
-	ResultRequeue           = common.ResultRequeue
-	ResultRequeueAfter5mins = common.ResultRequeueAfter5mins
-	MetricResType           = common.MetricResTypeVPC
+	log           = logger.Log
+	MetricResType = common.MetricResTypeNetworkInfo
 )
 
-// VPCReconciler VPCReconcile reconciles a VPC object
-type VPCReconciler struct {
+// NetworkInfoReconciler NetworkInfoReconcile reconciles a NetworkInfo object
+// Actually it is more like a shell, which is used to manage nsx VPC
+type NetworkInfoReconciler struct {
 	Client   client.Client
 	Scheme   *apimachineryruntime.Scheme
 	Service  *vpc.VPCService
 	Recorder record.EventRecorder
 }
 
-func (r *VPCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	obj := &v1alpha1.VPC{}
-	log.Info("reconciling VPC CR", "VPC", req.NamespacedName)
-	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeVPC)
+func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	obj := &v1alpha1.NetworkInfo{}
+	log.Info("reconciling NetworkInfo CR", "NetworkInfo", req.NamespacedName)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeNetworkInfo)
 
 	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
-		log.Error(err, "unable to fetch VPC CR", "req", req.NamespacedName)
+		log.Error(err, "unable to fetch NetworkInfo CR", "req", req.NamespacedName)
 		return common.ResultNormal, client.IgnoreNotFound(err)
 	}
 
 	if obj.ObjectMeta.DeletionTimestamp.IsZero() {
-		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeVPC)
-		if !controllerutil.ContainsFinalizer(obj, commonservice.VPCFinalizerName) {
-			controllerutil.AddFinalizer(obj, commonservice.VPCFinalizerName)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeNetworkInfo)
+		if !controllerutil.ContainsFinalizer(obj, commonservice.NetworkInfoFinalizerName) {
+			controllerutil.AddFinalizer(obj, commonservice.NetworkInfoFinalizerName)
 			if err := r.Client.Update(ctx, obj); err != nil {
-				log.Error(err, "add finalizer", "VPC", req.NamespacedName)
+				log.Error(err, "add finalizer", "NetworkInfo", req.NamespacedName)
 				updateFail(r, &ctx, obj, &err, r.Client)
 				return common.ResultRequeue, err
 			}
-			log.V(1).Info("added finalizer on VPC CR", "VPC", req.NamespacedName)
+			log.V(1).Info("added finalizer on NetworkInfo CR", "NetworkInfo", req.NamespacedName)
 		}
 
-		createdVpc, nc, err := r.Service.CreateorUpdateVPC(obj)
+		createdVpc, nc, err := r.Service.CreateOrUpdateVPC(obj)
```

Comment:
CreateOrUpdateAVIRule should be skipped if createdVpc returned from CreateOrUpdateVPC is a shared VPC(not created for this ns), check IsSharedVPCNamespaceByNS before call CreateOrUpdateAVIRule.

Reply:
fixed